modules/gapi/src/streaming/onevpl/engine/preproc/preproc_session.hpp

```diff
@@ -0,0 +1,53 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2021 Intel Corporation
+
+#ifndef GAPI_STREAMING_ONVPL_PREPROC_SESSION_HPP
+#define GAPI_STREAMING_ONVPL_PREPROC_SESSION_HPP
+#include <memory>
+#include <queue>
+
+#include <opencv2/gapi/streaming/meta.hpp>
+#include "streaming/onevpl/engine/engine_session.hpp"
+#include "streaming/onevpl/accelerators/accel_policy_interface.hpp"
+#include "streaming/onevpl/engine/preproc/vpp_preproc_defines.hpp"
+
+#ifdef HAVE_ONEVPL
+#ifdef HAVE_INF_ENGINE
```

Comment:
currently PP is a part of IE backend because it depends from `InputInfo` at now. When OV2.0 would appeared (or other inference framework?) it could be simply expanded to other engines by wrapping `InputInfo` into generic structure

Reply:
So this will be unguarded when something else appears. OK